### ISSUE TITLE
SchemaGenerator throws SwaggerGeneratorException if model contains Dictionary with key of type "enum"

### ISSUE THREAD

**Issue by NickNilga** [domaindrivendev/Swashbuckle.AspNetCore#2406]
Swashbuckle.AspNetCore.SwaggerGen v 6.3.0

SchemaGenerator.GenerateSchemaForMember throws SwaggerGeneratorException  "Failed to generate schema for type" if type has a property of type Dictionary with key of type "enum". The inner exception is NullReferenceException is thrown from here because AdditionalProperties is null:

![image](https://user-images.githubusercontent.com/3526449/164174054-fb9942a7-f86f-480f-a5d6-e6c010e4197e.png)

Steps to reproduce:

1. Create model containing a property of type Dictionary with key of type enum. In my case it was:

`       [Serializable]
	[JsonConverter(typeof(StringEnumConverter))]
	[TypeConverter(typeof(EnumMemberConverter<SearchResultEntityType>))]
	public enum SearchResultEntityType
	{
		Unknown = 0,

		[EnumMember(Value = "PRODUCT")]
		Product = 1,

		[EnumMember(Value = "ACCOUNT")]
		Account = 2,

		[EnumMember(Value = "TICKET")]
		Ticket = 3,

		[EnumMember(Value = "INVOICE")]
		Invoice = 4,

		[EnumMember(Value = "PHONE_NUMBER_RESERVED")]
		PhoneNumberReserved = 5,

		[EnumMember(Value = "SIM_CARD_INACTIVE")]
		SimCardInactive = 6,

		[EnumMember(Value = "ORDER")]
		Order = 7,

		[EnumMember(Value = "ALL")]
		All = 8
	}`
	
	[Serializable]
	public class SearchPaginationResult : PaginationResult<SearchResultEntity>
	{
		/// <summary>
		/// Number of found entities in each search category
		/// </summary>
		[JsonProperty("searchCount", Order = 40)]
		public Dictionary<SearchResultEntityType, int> SearchCount { get; set; }
	}
2. Create endpoint that exposes type through API.
3. Open Swagger UI

**Comment by vovakuznetsov** [comment]
I think this behavior is because the AdditionalProperties object is not created for dictionaries with a given and limited set of keys.

![image](https://user-images.githubusercontent.com/101116262/164689561-78ffa088-a9b5-410d-8b5d-cb1134f4ffd3.png)

I was able to recreate the case in the test.
```
[Fact]
public void GenerateSchema_GenerateNestedDictionarySchema_IfNestedDictionaryTypeHasEnumKey()
{
  var schema = Subject().GenerateSchema(typeof(NestedDictionary), new SchemaRepository());
}
```
Where NestedDictionary is

```
using System.Collections.Generic;

namespace Swashbuckle.AspNetCore.TestSupport.Fixtures
{
    public class NestedDictionary
    {
        public Dictionary<IntEnum, object> DictionaryObject { get; set; }
    }
}
```

Explicitly creating AdditionalProperties helps to solve this problem, but unfortunately this is my first issue and I don't know if this solution is correct.

![image](https://user-images.githubusercontent.com/101116262/164693022-888ddbe0-dfa6-4949-ad9e-4df5d11032e8.png)

**Comment by JakubHolovsky** [comment]
Having the same issue, my class:

![image](https://user-images.githubusercontent.com/4558405/164695660-ea2d90eb-2738-4974-afb1-8403853adb30.png)

happens for me also since Swashbuckle.AspNetCore.SwaggerGen v 6.3.0 (so keeping the previous version for now)

**Comment by vovakuznetsov** [comment]
Duplicate  #https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/2368
MR #https://github.com/domaindrivendev/Swashbuckle.AspNetCore/pull/2373

**Comment by vitaliy-orlov** [comment]
The problem still happens on version 6.4.0

**Comment by fccoryan** [comment]
Seems fixed in 6.5.0
